framework/src/vectorpostprocessors/MaterialVectorPostprocessor.C

```diff
@@ -0,0 +1,143 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "MaterialVectorPostprocessor.h"
+#include "Material.h"
+#include "IndirectSort.h"
+
+#include "libmesh/quadrature.h"
+
+#include <numeric>
+
+template<>
+InputParameters validParams<MaterialVectorPostprocessor>()
+{
+  InputParameters params = validParams<ElementVectorPostprocessor>();
+  params.addRequiredParam<std::string>("material", "Material for which all properties will be recorded.");
```

Comment:
Use "MaterialName" as the type, it will allow Peacock/atom to give you a list of available materials in your input file.